packages/@aws-cdk/aws-appmesh/README.md

```diff
@@ -275,6 +281,78 @@ const gateway = new appmesh.VirtualGateway(this, 'gateway', {
   })],
   virtualGatewayName: 'gateway',
 });
+
+// A Virtual Gateway with listener TLS from a SDS provided certificate
+const gateway2 = new appmesh.VirtualGateway(this, 'gateway2', {
+  mesh: mesh,
+  listeners: [appmesh.VirtualGatewayListener.grpc({
+    port: 8080,
+    tls: {
+      mode: appmesh.TlsMode.STRICT,
+      certificate: appmesh.TlsCertificate.sds({
+        secretName: 'secrete_certificate',
+      }),
+    },
+  })],
+  virtualGatewayName: 'gateway2',
+});
+```
+
+## Adding mutual TLS authentication
+
+Mutual TLS authentication is an optional component of TLS that offers two-way peer authentication. To enable mutual TLS authentication, add `certificate` property to TLS Client Policy and/or `validation` property to TLS Listener.
+
+`tls.validation` and `tlsClientPolicy.certificate` can be sourced from either:
+
+- A customer provided certificate can be specified with a `certificateChain` path file and a `privateKey` file path.
+
+- A certificate provided by a Secrets Discovery Service (SDS) endpoint over local Unix Domain Socket can be specified with its `secretName`.
+
+```typescript
+import * as certificatemanager from '@aws-cdk/aws-certificatemanager';
+
+const node1 = new appmesh.VirtualNode(stack, 'node1', {
+  mesh,
+  serviceDiscovery: appmesh.ServiceDiscovery.dns('node'),
+  listeners: [appmesh.VirtualNodeListener.grpc({
+    port: 80,
+    tls: {
+      mode: appmesh.TlsMode.STRICT,
+      certificate: appmesh.TlsCertificate.file({
+        certificateChainPath: 'path/to/certChain',
+        privateKeyPath: 'path/to/privateKey',
+      }),
+      // Validate a file client certificates to enable mutual TLS authentication when a client provides a certificate.
+      validation: {
+        trust: appmesh.TlsValidationTrust.file({
+          certificateChain: 'path-to-certificate',
+        }),
+      },
+    },
+  })],
+});
+
+const node2 = new appmesh.VirtualNode(stack, 'node2', {
+  mesh,
+  serviceDiscovery: appmesh.ServiceDiscovery.dns('node2'),
+  backendDefaults: {
+    tlsClientPolicy: {
+      // Provide a SDS client certificate when a server requests it and enable mutual TLS authentication.
+      certificate: appmesh.TlsCertificate.sds( {
+        secretName: 'secret_certificate',
+      }),
+      ports: [8080, 8081],
+      validation: {
+        subjectAlternativeNames: {
+          exactMatch: ['mesh-endpoint.apps.local'],
+        },
+        trust: appmesh.TlsValidationTrust.acm({
+          certificateAuthorities: [acmpca.CertificateAuthority.fromCertificateAuthorityArn(stack, 'certificate', certificateAuthorityArn)],
+        }),
+      },
+    },
+  },
```

Comment:
I actually think we should change the order, so that the mutual TLS configuration is at the end here:

```suggestion
  backendDefaults: {
    tlsClientPolicy: {
      ports: [8080, 8081],
      validation: {
        subjectAlternativeNames: {
          exactMatch: ['mesh-endpoint.apps.local'],
        },
        trust: appmesh.TlsValidationTrust.acm({
          certificateAuthorities: [acmpca.CertificateAuthority.fromCertificateAuthorityArn(stack, 'certificate', certificateAuthorityArn)],
        }),
      },
      // Provide a SDS client certificate when a server requests it and enable mutual TLS authentication.
      certificate: appmesh.TlsCertificate.sds( {
        secretName: 'secret_certificate',
      }),
    },
  },
```

Reply:
I agree. This is addressed.